responsibleai/model_monitor/components/src/calculate_attribution.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""This file contains the core logic for feature attribution drift component."""
+import numpy as np
+import pandas as pd
+import logging
+
+from responsibleai import RAIInsights
+from sklearn.metrics import ndcg_score
+from ml_wrappers.model.predictions_wrapper import (
+    PredictionsModelWrapperClassification,
+    PredictionsModelWrapperRegression)
+
+try:
+    from lightgbm import LGBMClassifier, LGBMRegressor
+except ImportError:
+    pass
+
+from tabular.components.src._telemetry._loggerfactory import _LoggerFactory
+
+_logger = logging.getLogger(__file__)
+_ai_logger = None
+
+
+def _get_logger():
+    global _ai_logger
+    if _ai_logger is None:
+        _ai_logger = _LoggerFactory.get_logger(__file__)
+    return _ai_logger
+
+
+_get_logger()
+
+
+def create_lightgbm_model(X, y, task_type):
+    """Create model on which to calculate feature importances
+
+    Args:
+      X: list of x values (data excluding target columns)
+      y: list y values (target column data)
+      task_type: str, the task type (regression or classification) of the resulting model
+
+    Returns:
+       An appropriate model.
+    """
+    if (task_type == "classification"): 
+        lgbm = LGBMClassifier(boosting_type='gbdt', learning_rate=0.1,
+                              max_depth=5, n_estimators=200, n_jobs=1, random_state=777)
+    else: 
+        lgbm = LGBMRegressor(boosting_type='gbdt', learning_rate=0.1,
+                            max_depth=5, n_estimators=200, n_jobs=1, random_state=777)
+    model = lgbm.fit(X, y)
+    
+    _logger.info("Created lightgbm model")
+    return model
+
+
+def get_model_wrapper(task_type, target_column, baseline_df, production_df):
+    """Create model wrapper using ml-wrappers on which to calculate feature importances
+
+    Args:
+      task_type: str, The task type (regression or classification) of the resulting model
+      target_column: str, the column to predict
+      baseline_df: The baseline dataframe
+      production_df: The production dataframe
```

Comment:
I'm guessing "production" refers to data that came in while monitoring? What is "baseline"? A previous time period's data? Would be nice to have a brief comment I think.